Content.Shared/StationRecords/SharedGeneralStationRecordConsoleSystem.cs

```diff
@@ -31,15 +35,18 @@ public sealed class GeneralStationRecordConsoleState : BoundUserInterfaceState
     public StationRecordKey? SelectedKey { get; }
     public GeneralStationRecord? Record { get; }
     public Dictionary<StationRecordKey, string>? RecordListing { get; }
-
-    public GeneralStationRecordConsoleState(StationRecordKey? key, GeneralStationRecord? record, Dictionary<StationRecordKey, string>? recordListing)
+    public GeneralStationRecordsFilter? Filter { get; }
+    public GeneralStationRecordConsoleState(StationRecordKey? key, GeneralStationRecord? record,
+        Dictionary<StationRecordKey, string>? recordListing, GeneralStationRecordsFilter? newFilter)
     {
         SelectedKey = key;
         Record = record;
         RecordListing = recordListing;
+        Filter = newFilter;
     }
 
-    public bool IsEmpty() => SelectedKey == null && Record == null && RecordListing == null;
+    public bool IsEmpty() => SelectedKey == null
+        && Record == null && RecordListing == null && Filter == null;
```

Comment:
```suggestion
        && Record == null && RecordListing == null;
```

`IsEmpty` is supposed to be used when the records database is empty